### DISCUSSION TITLE
Background Sync not working

### DISCUSSION THREAD

**Discussion by Proxymiity** [MobiusSync/MobiusSync#154] [Q&A] [answered]
I've been using the app for two weeks now, background sync enabled (both in the settings, set to "Wi-Fi & Cellular", and in the app-specific settings). However, background sync never happens (the app itself shows "none" and the other devices show no sign of connection).

The FAQ says that it could take up to 24 hours for background sync to start.
In the app, I've set the power sync to a 10min interval, and the battery sync to a 15min interval.

Device: iPhone 15 Pro Max
iOS version: 17.3.1
App version: 1.23.2
3 folders in Syncthing, 1GB in total size (if that helps)

Here's a screenshot of the Syncthing UI:
![image](https://github.com/MobiusSync/MobiusSync/assets/25194496/2ec95875-44e1-450c-9e00-04c28c58ee68)

And screenshots of the settings:
![image](https://github.com/MobiusSync/MobiusSync/assets/25194496/c05f5909-9fde-4f5c-af87-b4b9a2148aa4)
![image](https://github.com/MobiusSync/MobiusSync/assets/25194496/0644ba82-c2d7-4c29-a077-8b8cfa5c7a6c)

**Comment by Proxymiity** [comment]
Just tested that background sync is indeed working after a day on an iPhone 13 running iOS 17.4.1. 
I might have a 11 Pro running 17.3, will try to see if this is specific to 17.3.

**Comment by Proxymiity** [comment]
Update: The 11 Pro running 17.3.1 did not sync in the background. Neither did the iPad on 17.1.
Upgrading both phones to 17.4.1 fixed the background sync issue. Will test on the iPad and update the issue after.

@SimonPickup is this an expected behavior?

**Comment by danielraffel** [comment]
I'm experiencing somewhat similar behavior. App works fine on iPhone but on iPad it does not. The iPad will occasionally background sync then stop and only way to get it working is to launch the app. I've tried deleting and reinstalling the app and background seems to work for a short bit then stops. I dunno if it's related but keeping notifications enabled might help...not sure cuz eventually the iPad stops syncing in the background. Very frustrating.

**Comment by trigger2k20** [comment]
@Proxymiity regarding "Will test on the iPad and update the issue after." How went the testing ? 

Thinking of switching from Resilio sync to syncthing (Macs/Linux)

**Comment by Proxymiity** [comment, marked as answer]
Completely forgot to update the issue. My bad.

It appears that the iOS task scheduler is somewhat funky. If the device reboots, you have to open the app at least once to get background sync to work. Then, it'll sync in the background at regular intervals for a week, sometimes two. Then you have to restart it, and so on.

I find that it works the best when the device is on the charger (at least on the iPad -- maybe that's a cellular thing?).
Maybe we could just set up a shortcut automation to open the app every day at midnight or when the device gets plugged in?

In the meantime, I'm just opening the app once in a while, and it gets the job done.

**Reply by trigger2k20** [reply]
Same behavior with Resilio Sync. Maybe both should talk with Apple....

**Comment by Proxymiity** [comment]
> Thinking of switching from Resilio sync to syncthing (Macs/Linux)

On macOS you just need to install the official Syncthing app and it'll work right away. Same goes for Linux devices.

**Comment by trigger2k20** [comment]
Thanks @Proxymiity , on Resilio Sync side, the App is officially not syncing because of the quirks of iOS. Reading your answer, I can understand why. So if I working with OS365 via my cloud I have to open Resilio Sync  manually to sync which is true 1990's. Maybe your suggested "shortcut automation" would work but how to setup...

PS: Just asking, can I define a white/blacklist via Synching GUI ? In Resilio you have to edit a file....

**Comment by Proxymiity** [comment]
Well, well, background sync is working as we speak. 
![image](https://github.com/MobiusSync/MobiusSync/assets/25194496/b5b220a9-b3bc-43cb-8506-e528b5347a7c)

Regarding the blacklist, are you talking about synced files? Or are you talking about the clients you are syncing to?

**Comment by trigger2k20** [comment]
@Proxymiity thanks, i am talking about files.

**Comment by Proxymiity** [comment]
> @Proxymiity thanks, i am talking about files.

Syncthing has a .stignore file format. You can edit it straight in the interface, and it's pretty straightforward

Here's an example of stignore I use:
```
(?d).DS_Store
(?d)venv
(?d).venv
(?d)node_modules
(?d).git
(?d).idea
(?d)@eaDir
(?d)__pycache__
```
The `(?d)` means that the ignored directory does not prevent removal of a parent, if any. 

You can have a look at the docs here: https://docs.syncthing.net/v1.23.2/users/ignoring

**Comment by trigger2k20** [comment]
Oh it looks better, I give it a try, thanks !
